File: ios/GlobalConfig.swift
//
//  GlobalConfig.swift
//  react-nav-page
//
//  Created by Yusuf Zeren on 26.04.2024.
//

struct GlobalConfig {
    static var headerShow:Bool = true
    static var headerTransparent:Bool = false
    static var headerBackgroundColor:String = "#9c27b0"
    static var hederNavBarAlpha:Float = 1.0
    
    
    static func getConfig() -> [String: Any] {
        return [
            "headerShow": headerShow,
            "headerTransparent": headerTransparent,
            "headerBackgroundColor":headerBackgroundColor,
            "hederNavBarAlpha":hederNavBarAlpha,
        ]
    }
    
    static func setConfig(config: [String: Any]) {
        if let headerShowValue = config["headerShow"] as? Bool {
            headerShow = headerShowValue
        }
        if let headerTransparentValue = config["headerTransparent"] as? Bool {
            headerTransparent = headerTransparentValue
        }
        if let headerBackgroundColorValue = config["headerBackgroundColor"] as? String {
            headerBackgroundColor = headerBackgroundColorValue
        }
        if let hederNavBarAlphaValue = config["hederNavBarAlpha"] as? Float {
            hederNavBarAlpha = hederNavBarAlphaValue
        }
    }
}

File: android/src/main/java/com/reactnavpage/GlobalConfig.kt
package com.reactnavpage

import com.facebook.react.bridge.WritableMap
import com.facebook.react.bridge.WritableNativeArray
import com.facebook.react.bridge.WritableNativeMap


object GlobalConfig {
  var hederNavBarAlpha: Double = 1.0
  var headerShow: Boolean = true
  var headerTransparent: Boolean = false
  var headerBackgroundColor:String = "#9c27b0"

  fun getConfig(): HashMap<String, Any> {
    return hashMapOf(
      "hederNavBarAlpha" to hederNavBarAlpha,
      "headerShow" to headerShow,
      "headerTransparent" to headerTransparent,
      "headerBackgroundColor" to headerBackgroundColor
    )
  }

  fun setConfig(config: HashMap<String, Any>) {
    hederNavBarAlpha = config["hederNavBarAlpha"] as? Double ?: hederNavBarAlpha
    headerShow = config["headerShow"] as? Boolean ?: headerShow
    headerTransparent = config["headerTransparent"] as? Boolean ?: headerTransparent
    headerBackgroundColor = config["headerBackgroundColor"] as? String ?: headerBackgroundColor
  }
}

fun getConfigMap(): WritableNativeMap {
  val config = WritableNativeMap()
  config.putDouble("hederNavBarAlpha", GlobalConfig.hederNavBarAlpha)
  config.putBoolean("headerShow", GlobalConfig.headerShow)
  config.putBoolean("headerTransparent", GlobalConfig.headerTransparent)
  config.putString("headerBackgroundColor", GlobalConfig.headerBackgroundColor)

  return config
}

